### ISSUE TITLE
Making confirm on customized dataset

### ISSUE THREAD

**Issue by tegg89** [thtrieu/darkflow#416]
Thanks for sharing this wonderful codes.
I would like to detect customized dataset and searched instruction on `readme` file and pre-questioned issues. 

First, I have created new `cfg` w.r.t. original file. In my case, only one target label is needed, so I edited `label.txt` and corresponded parameters in new `cfg`.
When I trained this new `cfg` file with the customized dataset, I cannot detect any object, even training dataset.
Afterward, I thought using pre-trained weights for capturing low-level features of objects and then train with my dataset.

In my knowledge, transfer learning is to freeze all layers except last few layers. So I changed codes to freeze layers and trained with my dataset. When testing, I have to release layers and feed-forward from top to bottom. At this case, trained weights only cover last few layers, causing an error about not matched sizes of weights.

Therefore, I have to release all layers and trained as below:
`flow --model yolo-new.cfg --load yolo.weights --dataset <new-dataset> --annotation <new-annotation> --train --gpu 1.0`
It means that training new dataset on the pre-trained weights. I think this is continual learning (domain adaptation), not transfer learning.

At the result, new objects can be detected well.

TL;DR: 
1. Confirm for theoretical knowledge about twos:
Transfer learning: freeze layers at training time and release all layers at testing time
Continual learning: try to train new label upon pre-trained dataset
2. Confirm for whether above command is continual learning or not

**Comment by srafay** [comment]
Are you still facing this problem?
@tegg89

**Comment by tegg89** [comment]
@srafay I already have results but would like to make confirmation of theoretical understanding from what I did. I am confusing above method is whether transfer learning or continual learning (or domain adaptation).

**Comment by EnricoBeltramo** [comment]
I had similar issue: after a long training with a custom model, it wasn't able to detect anything. I try again loading from presaved weights as tegg99 suggest. There are some error or workaround about that?

**Comment by tegg89** [comment]
@EnricoBeltramo : I prefer to use pre-trained weights. If the result is not satisfied, try to increase the size of the dataset or that of the model. Look for 'machine learning recipe' by Andrew Ng.

**Comment by EnricoBeltramo** [comment]
I work with a limited resource system (a Raspebby pi 3) and this system is able to manage only tiny-yolo in a reasonable time. Unfortunately tiny-yolo is not so perfect as detection, so my hope way that training the network for some few class (in my case 4) should be improve the performance. Doing the training how explained before anyway I find that the dectection is generally worse and I don't understand why (my expectation was that it work at least as before). I trained using VOCtrainval_11-May-2012 dataset using follow commands:
python flow --model cfg/tiny-yolo-voc-4c.cfg --load bin/tiny-yolo-voc.weights --dataset "path dataset" --annotation "path Annotations" --train --gpu 1.0
I stopped at step 3313, with follow result:
step 3313 - loss 1.9788038730621338 - moving ave loss 2.77185750016443
saved last checkpoint in protobuf:
python flow --model cfg\tiny-yolo-voc-4c.cfg --load -1 --savepb
And tested with folder tests:
python flow --pbLoad built_graph/tiny-yolo-voc-4c.pb --metaLoad built_graph/tiny-yolo-voc-4c.meta --imgdir sample_img/
Of all example in the folder, it was able to detect only a person in sample_person.jpg (the classes that I work on are: person, car, cat, dog)
